version: '3.8'
services:
  authservice: &authservice
    build:
      context: ../dapz-auth-py-flask
      dockerfile: Dockerfile.dev
    env_file:
      - ../dapz-auth-py-flask/envs/.env.development
    environment:
      - FLASK_ENV=development
      - FLASK_APP=app.py
    stdin_open: true
    tty: true
    depends_on:
      - postgres
    networks:
      - dapperzen
    volumes:
      - ../dapz-auth-py-flask:/app
    command: ["python", "-m", "flask", "run", "--host=0.0.0.0", "--port=5000"]
    ports:
      - "5000:5000"

  authservice-debug:
    <<: *authservice
    ports:
      - "5678:5678"
    command: ["python", "-m", "debugpy", "--listen", "0.0.0.0:5678", "--wait-for-client", "-m", "flask", "run", "--host=0.0.0.0", "--port=5000"]

  authservice-test:
    <<: *authservice
    ports:
      - "5678:5678"
    command: ["pytest", "-v", "--cov"]

  dapzaccessguard: &dapzaccessguard
    build:
      context: ../dapz-access-guard-py-flask
      dockerfile: Dockerfile.dev
    env_file:
      - ../dapz-access-guard-py-flask/envs/.env.development
    stdin_open: true
    tty: true
    depends_on:
      - postgres
    networks:
      - dapperzen
    volumes:
      - ../dapz-access-guard-py-flask:/app
    command: ["python", "setup.py", "bdist_wheel"]

  dapzaccessguard-test:
    <<: *dapzaccessguard
    env_file:
      - ../dapz-access-guard-py-flask/envs/.env.testing
    ports:
      - "5678:5678"
    entrypoint: ["python" ,"-m"]
    command: ["pytest", "-vv", "--cov=.", "--cov-report=term-missing"]